public/app/features/dashboard-scene/panel-edit/VizPanelManager.tsx

```diff
@@ -97,6 +101,27 @@ export class VizPanelManager extends SceneObjectBase<VizPanelManagerState> {
       variables = gridItem.parent.state.$variables.clone();
     }
 
+    if (repeatOptions.repeat) {
+      const variable = sceneGraph.lookupVariable(repeatOptions.repeat, gridItem);
+
+      if (variable instanceof MultiValueVariable && variable.state.value.length) {
+        const { values, texts } = getMultiVariableValues(variable);
+
+        const varWithDefaultValue = new LocalValueVariable({
+          name: variable.state.name,
+          value: values[0],
+          text: String(texts[0]),
+        });
+        if (!variables) {
+          variables = new SceneVariableSet({
+            variables: [varWithDefaultValue],
+          });
+        } else {
+          variables.state.variables.push(varWithDefaultValue);
```

Comment:
Should we do `variables.setState({variables})` here? It won't trigger re-render in components subscribed to this state. Is this on purpose?

Reply:
Yeah, it should use setState, thank you!